### PR TITLE
DP-20411 Add data fields for binder content type

### PR DESCRIPTION
**Pull request by ygannett**
**Description:**
- Set up data fields to binder content type:
     - Data flag
     - Data type
     - Data resource type
     - Data format
- Set up the conditions to display the fields (Data type, Data resource type, Data format) based on the state of Data flag.
- Add **Mass Metatag** to the binder meta configuration.
- Output the meta data as shown in the test script below.

**Jira:** (Skip unless you are MA staff)
https://jira.mass.gov/browse/DP-20411

**To Test:**
- [ ] Go to any binder edit page or create new binder.
- [ ] Scroll down the overview tab to the Organization(s).
- [ ] Find the Data flag field between the Organization(s) and the Intended Audience.

flag-unchecked

- [ ] Check the Include this page in the “Data” tab within search, and find the rest of the data fields are displayed.
flag-checked

- [ ] Make selection to test for meta data output with the added field data:

### \<meta name="category">

|   | meta output  | field\_data\_flag | field\_binder\_binder\_type | note |
| ---:|:----------|:-------------:|:-------------:|:-------|
|  1  | no output    | unchecked        | Archive |       |
|  2  | \<meta name="category" content="data"\> | checked | Archive | |
|  3  | no output    | unchecked   | Audit | |
|  4  | \<meta name="category" content="data"\> | checked | Audit | |
|  5  | no output    | unchecked   | Exam Results | |
|  6  | \<meta name="category" content="data"> | checked | Exam Results | |
|  7  | no output    | unchecked   | Handbook | |
|  8  | \<meta name="category" content="data"> | checked | Handbook | |
|  9  | \<meta name="category" content="laws-regulations">    | unchecked   | Law Library | This is the original implementation. No edit in this ticket. |
| 10  | \<meta name="category" content="laws-regulations"> | checked | Law Library | This is the original implementation. No edit in this ticket. |
|  11 | no output    | unchecked   | Manual | |
|  12 | \<meta name="category" content="data"> | checked | Manual | |
|  13 | no output    | unchecked   | Report | |
|  14 | \<meta name="category" content="data"> | checked | Report | |
|  15 | no output    | unchecked   | Resource | |
|  16 | \<meta name="category" content="data"> | checked | Resource |  |
|  17 | no output    | unchecked   | Tool Kit | |
|  18 | \<meta name="category" content="data"> | checked | Tool Kit | |

### \<meta name="mg_type">
~~Note: The first meta value is from field\_binder\_binder\_type as the original mg_type value. It's shown as the first value.  Below sample's field\_binder\_binder\_type value is **resource**.  If your test sample content would output its selected value.~~

- Override the original value of "resource" with the field value.
- field\_data\_resource\_type value doesn't affect the content value.

|   | meta output  | field\_data\_resource\_type | field\_binder\_data\_type | note |
|:--:|:-------------|:-------------:|:-------------:|:-----|
|  1  | ~~\<meta name="mg_type" content="resource">~~ No output | unchecked | N/A | |
|  2  | ~~\<meta name="mg_type" content="resource">~~ No output | checked   | N/A | |
|  3  | ~~\<meta name="mg_type" content="resource, **Data resource**">~~ \<meta name="mg_type" content="data-resource"> | unchecked | Data resource | ~~This value needs to be confirmed. No spec in the ticket with this pattern.~~ |
|  4  | ~~\<meta name="mg_type" content="resource, report">~~ \<meta name="mg_type" content="data-resource"> | checked   | Data resource | ~~populated from field\_dataresource\_metatag~~ |
|  5  | ~~\<meta name="mg_type" content="resource, dataset">~~ \<meta name="mg_type" content="dataset"> | unchecked | Dataset | |
|  6  | ~~\<meta name="mg_type" content="resource, dataset">~~ \<meta name="mg_type" content="dataset"> | checked   | Dataset | |
|  7  | ~~\<meta name="mg_type" content="resource, data-catalog">~~ \<meta name="mg_type" content="data-catalog"> | unchecked | Data catalog | |
|  8  | ~~\<meta name="mg_type" content="resource, data-catalog">~~ \<meta name="mg_type" content="data-catalog"> | checked   | Data catalog | |

### [NEWLY ADDED] \<meta name="mg_resource_type">
Output **field\_data\_resource\_type** value.

Currently it has only 1 options, but all selected options are added separated by ",":

\<meta name="mg_resource_type" content="report">

If no option is selected, **null** is output:

\<meta name="mg_resource_type" content="null">

### \<meta name="mg_encoding_format">
All selected options are added to the content:

\<meta name="mg_encoding_format" content="csv,doc,gdb,geojson">

**Screenshots/GIFs:**
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

**Comment by clairesunstudio** [comment]
questions and comments:

1. I suggested some edits to the descriptions of the fields --> https://docs.google.com/document/d/1ejHFNmTNJeG8HK1vbbOIGXEVXLeWxHavZhLDw_NYvio/edit# 
2. I think when the user selects law library, the `type` and `resource-type` should not exist or be set to`null`
Screen Shot 2021-03-18 at 6 07 14 PM
3. When I select a different data type from `data-resource`, the `resource type` value still persists. It should not exist or be set to `null`. e.g. 
Screen Shot 2021-03-18 at 6 22 11 PM
^ When subtype/binder type is set to report, should that affect data type? I think this can be confusing to authors.

**Comment by clairesunstudio** [comment]
Talked with Joseph Galluccio about my comments:

1. Ignore the text edits in this ticket, we will create a separate ticket to address those across all data content types. I will reach out to Kristina Johnson and get her input on those edits. 
2. For the law library binder type, pushed up a change to hide the data fields --> a3d1896
3. For the invisible field set to "null" values, @ygannett can you take a look at those and see how hard is it to remove values from hidden fields? 

Otherwise LGTM, and this is ready for internal review.

File: conf/drupal/config/metatag.metatag_defaults.node__binder.yml [hunk 1]
```diff
@@ -5,16 +5,14 @@ dependencies: {  }
...
+  mass_metatag_resource_type: '[node:field_binder_binder_type:entity:name]'
+  mass_metatag_short_description: '[node:field_binder_short_desc]'
   mg_stakeholder_org: '[node:field_state_organization_tax:entity:name]'
-  mass_metatag_associated_organization: '[node:field_binder_ref_organization]'
-  mass_metatag_type: '[node:field_binder_binder_type:entity:name]'
   mass_metatag_sub_type: '[node:field_binder_binder_type:entity:name]'
-  mass_metatag_contact_details: '[node:field_contact]'
+  mass_metatag_type: '[node:field_binder_binder_type:entity:name]'
   description: '[node:field_binder_short_desc]'
-  og_description: '[node:field_binder_short_desc]'
```
Comment (arthurbaghdas) [inline conf/drupal/config/metatag.metatag_defaults.node__binder.yml:18]: I think we should't remove this.

**Review by arthurbaghdas — CHANGES_REQUESTED** [review]

**Comment by arthurbaghdas** [comment]
Test is failing. I think due to metatag missing configuration.

File: conf/drupal/config/metatag.metatag_defaults.node__binder.yml (hunk 1 above)
Comment (ygannett) [reply conf/drupal/config/metatag.metatag_defaults.node__binder.yml:18]: Thank you for pointing that out.  It was back on with some other config stripped off.

**Comment by ygannett** [comment]
@arthurbaghdas Can you review this again?

**Review by arthurbaghdas — APPROVED** [review]

**Pull request merged**